astro-airflow-iris/cookiecutter.json

```diff
@@ -1,6 +1,6 @@
 {
-    "project_name": "New Kedro Project",
-    "repo_name": "{{ cookiecutter.project_name.replace(' ', '-').lower().strip('-') }}",
-    "python_package": "{{ cookiecutter.project_name.replace(' ', '_').replace('-', '_').lower() }}",
+    "project_name": "Iris",
+    "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
+    "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}"
 }
```

Comment:
I think you've got it! Python packaging is so confusing 😀 

`project_name` is just a kedro concept and not very important. It's mainly used just in docs (e.g. README.md contains `{{ cookiecutter.project_name }}`). It also goes to `pyproject.toml`, which is loaded as `metadata` then is occasionally used as `metadata.project_name`. It's nothing to do with Python packaging.

`repo_name`: indeed, this doesn't really do anything apart from choose the name of the directory. It can be anything you like. It's also nothing to do with Python packaging.

`python_package`: yes, this is the name used in `import` name. As you say, in our case we also make it the distribution name since we set `name="{{ cookiecutter.python_package }}"` in setup.py. There's no rule about not using underscores in the distribution name, but at some point in setuptools/pip `_` always gets converted to `-` automatically.